app/src/main/java/woowacourse/paint/ui/custom/DrawingCanvas.kt

```diff
@@ -0,0 +1,165 @@
+package woowacourse.paint.ui.custom
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Paint
+import android.graphics.Path
+import android.util.AttributeSet
+import android.view.MotionEvent
+import android.view.View
+import androidx.annotation.ColorInt
+import woowacourse.paint.drawing.Drawing
+import woowacourse.paint.drawing.DrawingCircle
+import woowacourse.paint.drawing.DrawingHistory
+import woowacourse.paint.drawing.DrawingRectangle
+import woowacourse.paint.drawing.Eraser
+import woowacourse.paint.ui.PaintMode
+import woowacourse.paint.ui.PathPoint
+
+class DrawingCanvas @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null) :
+    View(context, attrs) {
+    private var drawing = Drawing(Path(), Paint())
+
+    private val drawingHistory = DrawingHistory()
+
+    private var paintMode = PaintMode.PEN
+
+    private val drawingRectangle = DrawingRectangle()
+    private val drawingCircle = DrawingCircle()
+    private val eraser = Eraser(drawingHistory)
+
+    init {
+        changePaintProperty(Color.RED, DEFAULT_PAINT_WIDTH)
+    }
+
+    override fun onDraw(canvas: Canvas) {
+        super.onDraw(canvas)
+        drawDrawingHistory(canvas)
+        when (paintMode) {
+            PaintMode.PEN -> canvas.drawPath(drawing.path, drawing.paint)
+            PaintMode.RECTANGLE, PaintMode.FILL_RECTANGLE -> drawingRectangle.drawShapeOnCanvas(
+                canvas, drawing.paint
+            )
+
+            PaintMode.CIRCLE, PaintMode.FILL_CIRCLE -> drawingCircle.drawShapeOnCanvas(
+                canvas,
+                drawing.paint
+            )
+
+            PaintMode.ERASER -> {}
```

Comment:
```suggestion
            PaintMode.ERASER -> Unit
```

아무것도 안해줄 땐 빈괄호보다 `Unit`으로 명시하는게 이쁩니다 😆

Reply:
하나만 있는줄 알고 해놨는데 여러개군요 ㅋㅋㅋ 베리가 잘 보시고 맘에 드신다면 다 바꿔주시고 아니라면 그냥 두셔도 좋습니다.

Reply:
오오!!! 좋네요 ㅎㅎ